geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/maps/package.scala

```diff
@@ -0,0 +1,12 @@
+package org.locationtech.geomesa.utils
+
+package object maps {
+
+  def aggregateMaps[K,V](maps: TraversableOnce[collection.Map[K,V]], operation: (V,V) => V, emptyVal: V): Map[K,V] =
+    maps.flatMap(_.toSeq).foldLeft(Map[K,V]()){ case (mapSoFar, (k, v)) =>
+      mapSoFar.updated(k, operation(mapSoFar.getOrElse(k, emptyVal), v))
+    }
```

Comment:
I think it would be more efficient to use a mutable map and add everything to it. Seems like this could create a lot of intermediate maps.